src/api.ts

```diff
@@ -68,14 +69,41 @@ export async function getNonceToSign({
   );
 }
 
+export async function exchangeToken({
+  baseUrl,
+  requirements,
+  accessToken,
+  ...options
+}: ExchangeTokenEndpointOptions) {
+  const body = {
+    ...requirements,
+  };
+  const queryString = createQueryParams(options);
+  return await getJSON<LoginWithSignedNonceResponse>(
+    `${baseUrl}/exchange_token?${queryString}`,
+    10000,
+    'default',
+    '',
+    {
+      method: 'POST',
+      body: JSON.stringify(body),
+      headers: {
+        Authorization: `Bearer ${accessToken}`,
+        'Content-Type': 'application/json',
+        'X-SlashAuth-Client': btoa(JSON.stringify(DEFAULT_SLASHAUTH_CLIENT)),
+      },
+    }
+  );
+}
+
 export async function loginWithSignedNonce({
   baseUrl,
   ...options
 }: LoginWithSignedNonceOptions) {
   const queryString = createQueryParams(options);
   return await getJSON<LoginWithSignedNonceResponse>(
     `${baseUrl}/loginWithSignedNonce?${queryString}`,
-    1000,
+    10000,
```

Comment:
would three seconds be enough?

Reply:
I figure let's go lnog for now -- we can start tracking analytics!